lib/THNN/generic/SpatialAveragePooling.c

```diff
@@ -74,7 +62,7 @@ static int nn_(SpatialAveragePooling_updateOutput)(lua_State *L)
     THTensor_(resize4d)(output, input->size[0], nInputPlane, outputHeight, outputWidth);
   
   input = THTensor_(newContiguous)(input);
-  luaL_argcheck(L, THTensor_(isContiguous)(output), 1, "");
+  THArgCheck(THTensor_(isContiguous)(output), 1, "");
```

Comment:
Good catch.
Will fix the error message (but could eventually call `newContiguous` as well) and the `THArgCheck` index (it was 1 because the `self` was the first argument before).
BTW, it's sometimes difficult to specify an index when several arguments are involved (like comparing the input with `kW`).